taskweaver/llm/util.py

```diff
@@ -1,6 +1,6 @@
 from typing import Dict, Literal, Optional
 
-ChatMessageRoleType = Literal["system", "user", "assistant"]
+ChatMessageRoleType = Literal["system", "user", "assistant", "function"]
```

Comment:
Have we used "function" role type currently when preparing prompt?

Reply:
no, only used to pass the output back to the code geneator in order to distinguish the two different types of response: 1. message, 2. function calling

Reply:
Maybe we should still follow the OpenAI schema as they are using "function" for serving the purpose of feeding the response after the function call generated. And the function call is still in "assistant" role. Ollama or other impls would tend to follow OpenAI's schema and we might need to do lots of conversion later

Reply:
https://json-schema.org/understanding-json-schema/reference/type